core/store/lib/search/adapter/api/searchAdapter.ts

```diff
@@ -170,5 +170,32 @@ export class SearchAdapter {
         return this.handleResult(resp, 'review', start, size)
       }
     })
+    this.registerEntityType('cms_page', {
+      queryProcessor: (query) => {
+        // function that can modify the query each time before it's being executed
+        return query
+      },
+      resultPorcessor: (resp, start, size) =>  {
+        return this.handleResult(resp, 'cms_page', start, size)
+      }
+    })
+    this.registerEntityType('cms_block', {
+      queryProcessor: (query) => {
+        // function that can modify the query each time before it's being executed
+        return query
+      },
+      resultPorcessor: (resp, start, size) =>  {
+        return this.handleResult(resp, 'cms_block', start, size)
+      }
+    })
+    this.registerEntityType('cms_hierarchy', {
+      queryProcessor: (query) => {
+        // function that can modify the query each time before it's being executed
+        return query
+      },
+      resultPorcessor: (resp, start, size) =>  {
+        return this.handleResult(resp, 'cms_hierarchy', start, size)
+      }
+    })
```

Comment:
These entities should be registered from the module level - keeping all the CMS oriented logic inside the cms module - not in the core I believe (currently CMS is also a module without references from the core)

Reply:
same as for gql comments - Lets move this update to another PR

Reply:
Could you create an issue for that?

Reply:
Issue is created